conbench/entities/history.py

```diff
@@ -143,7 +194,61 @@ def get_history(case_id: str, context_id: str, hardware_hash: str, repo: str) ->
         history_df, include_current_commit_in_rolling_stats=False
     )
 
-    return list(history_df.itertuples())
+    # return list(history_df.itertuples())
+
+    samples: list[HistorySample] = []
+    # Iterate over rows of pandas dataframe; get each row as namedtuple.
+    for sample in history_df.itertuples():
+        # Note(JP): the Commit.timestamp is nullable, i.e. not all Commit
+        # entities in the DB have a timestamp (authoring time) attached.
+        # However, in this function I believe there is an invariant that the
+        # query only returns Commits that we have this metadata for (what's the
+        # precise reason for this invariant? I think there is one). Codify this
+        # invariant with an assertion.
+        assert isinstance(sample.timestamp, datetime.datetime)
+
+        zstats = HistorySampleZscoreStats(
+            begins_distribution_change=sample.begins_distribution_change,
+            segment_id=sample.segment_id,
+            rolling_mean_excluding_this_commit=sample.rolling_mean_excluding_this_commit,
+            rolling_mean=_to_float(sample.rolling_mean),
+            residual=sample.residual,
+            rolling_stddev=_to_float(sample.rolling_stddev) or 0.0,
+        )
+
+        # For both, `sample.data` and `sample.times`, expect either None or a
+        # list. Make it so that in the output object they are always a list,
+        # potentially empty. `data` and `times` contain more than one value if
+        # this was a multi-sample benchmark.
+
+        data = []
+        if sample.data is not None:
+            data = [float(d) if d is not None else math.nan for d in sample.data]
+
+        times = []
+        if sample.times is not None:
+            times = [float(t) if t is not None else math.nan for t in sample.times]
```

Comment:
reading this code right now I am not even sure if `d` and `t` can be `None`. If this comes straight from the database then both `sample.data` and `sample.times` are either `None` or `postgresql.ARRAY(s.Numeric)` (see above). 

In that case, this should be fine:
```
[float(d) for d in sample.data]
```
because `d` is guarnateed to be a `decimal.Decimal()` and the following is fine:
```python
>>> float(Decimal('NaN'))
nan
```